src/types.jl

```diff
@@ -41,22 +42,25 @@ struct NewtonImmutableSolver{fType, algType, uType, resType, pType, INType, tolT
     tol::tolType
     cache::cacheType
     iip::Bool
+    prob::probType
 end
 
 # function NewtonImmutableSolver{iip}(iter, f, alg, u, fu, p, force_stop, maxiters, internalnorm, retcode, tol, cache) where iip
 #     NewtonImmutableSolver{iip, typeof(f), typeof(alg), typeof(u),
 #         typeof(fu), typeof(p), typeof(internalnorm), typeof(tol), typeof(cache)}(iter, f, alg, u, fu, p, force_stop, maxiters, internalnorm, retcode, tol, cache)
 # end
 
-struct BracketingSolution{uType}
+struct BracketingSolution{uType,resType}
```

Comment:
Are we still using the original solution type?

Reply:
Yep, we don't need to it. Refactored the code. Thanks for the suggestion.